[MTPG] Enable for some tests in test_fsdp_misc

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #102043

Enables MTPG for some FSDP tests in this file. Tests that need the
backward pass and warning logging are left as follow up work.

Backward pass issue: It seems that there is a hang with all_gather. Will sync with @kumpera on this.

Warning issue: We have a couple tests that regex check on warnings, but in the
multithreaded scenario these warnings are somehow not logged.

Differential Revision: [D43209769](https://our.internmc.facebook.com/intern/diff/D43209769/)

rohan-varma committed:
[MTPG] Enable for some tests in test_fsdp_misc

Enables MTPG for some FSDP tests in this file. Tests that need the
backward pass and warning logging are left as follow up work.

Backward pass issue: it seems that the registered MTPG is not recognized during
collectives in the backward pass, which makes sense since the collective is in
a different thread.

Warning issue: We have a couple tests that regex check on warnings, but in the
multithreaded scenario these warnings are somehow not logged.

Differential Revision: [D43209769](https://our.internmc.facebook.com/intern/diff/D43209769/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[MTPG] Enable for some tests in test_fsdp_misc"


Enables MTPG for some FSDP tests in this file. Tests that need the
backward pass and warning logging are left as follow up work.

Backward pass issue: It seems that there is a hang with all_gather. Will sync with kumpera on this.

Warning issue: We have a couple tests that regex check on warnings, but in the
multithreaded scenario these warnings are somehow not logged.

Differential Revision: [D43209769](https://our.internmc.facebook.com/intern/diff/D43209769/)

[ghstack-poisoned]

fduwjj commented:
Looks like some distributed tests failed?

rohan-varma committed:
Update on "[MTPG] Enable for some tests in test_fsdp_misc"


Enables MTPG for some FSDP tests in this file. Tests that need the
backward pass and warning logging are left as follow up work.

Backward pass issue: It seems that there is a hang with all_gather. Will sync with kumpera on this.

Warning issue: We have a couple tests that regex check on warnings, but in the
multithreaded scenario these warnings are somehow not logged.

Differential Revision: [D43209769](https://our.internmc.facebook.com/intern/diff/D43209769/)

[ghstack-poisoned]

awgu commented on test/distributed/fsdp/test_fsdp_misc.py:
```diff
@@ -184,46 +261,74 @@ def _check_equal(local, fsdp):
-        # computation.
-        class MyModel(nn.Module):
+    def test_fsdp_cpu_init_stays_on_cpu(self):
+        # Move me to MT test once warning logging and backward collective issue
```
Should we file an issue to track?

awgu reviewed (approved):
Lint error looks real :/

awgu reviewed (approved):
Lint error looks real :/

rohan-varma committed:
Update on "[MTPG] Enable for some tests in test_fsdp_misc"


Enables MTPG for some FSDP tests in this file. Tests that need the
backward pass and warning logging are left as follow up work.

Backward pass issue: It seems that there is a hang with all_gather. Will sync with kumpera on this.

Warning issue: We have a couple tests that regex check on warnings, but in the
multithreaded scenario these warnings are somehow not logged.

Differential Revision: [D43209769](https://our.internmc.facebook.com/intern/diff/D43209769/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[MTPG] Enable for some tests in test_fsdp_misc"


Enables MTPG for some FSDP tests in this file. Tests that need the
backward pass and warning logging are left as follow up work.

Backward pass issue: It seems that there is a hang with all_gather. Will sync with kumpera on this.

Warning issue: We have a couple tests that regex check on warnings, but in the
multithreaded scenario these warnings are somehow not logged.

Differential Revision: [D43209769](https://our.internmc.facebook.com/intern/diff/D43209769/)

[ghstack-poisoned]

rohan-varma commented:
@pytorchbot merge -f "CI done"
